server/v2/appmanager/service.go

```diff
@@ -12,6 +13,17 @@ import (
 	"cosmossdk.io/server/v2/core/transaction"
 )
 
+type Tx struct {
+	Tx         []byte // transaction bytes
+	Identifier string // transaction Identifier
+}
+
+type TxPool interface {
+	GetTxs(context.Context, uint32) ([]Tx, error)
+}
+
+type PrepareHandler func(ctx context.Context, txs []Tx) ([]Tx, []store.ChangeSet, error)
```

Comment:
So what is the idea of having this handler return a changeset? Specifically, 

- What will be done with the changeset?
- How does this handler read existing state?

Also, unrelated, `store.ChangeSet` should already contain all the change**s**, as it's a set. To have `[]store.ChangeSet` is a misnomer, as it reads, having a set of a set.

Reply:
there are two things to consider here. In the case of immediate execution we would need the state root returned as part of the block building process. In the case of optimistic execution we would need to return to comet and then wait for the change set. then on finalizeblock we write. This tries to accomidate those two paths that we will offer in the future